<fix>[compute]: support DiskAO.primaryStorageUuid for root volume

MatheMatrix opened the pull request:
If the DiskAO.primaryStoreUuid of the root disk is specified,
Directly use DiskAO.primaryStorageUuid as the primary storage
for the boot disk.

Resolves: ZSV-8585

Change-Id: I6c7a716373626779656b6c6b707864616d69706b

sync from gitlab !8339

Commit:
<fix>[compute]: support DiskAO.primaryStorageUuid for root volume

If the DiskAO.primaryStoreUuid of the root disk is specified,
Directly use DiskAO.primaryStorageUuid as the primary storage
for the boot disk.

Introduce VM allocation error.

Resolves: ZSV-8585

Change-Id: I6c7a716373626779656b6c6b707864616d69706b

zstack-robot-1 commented:
**Comment from wenhao.zhang:**

请保证 

LocalStorageDefaultAllocateCapacityFlow / LocalStorageDesignatedAllocateCapacityFlow

也同步修改

Commit:
<fix>[localstorage]: fix DiskAO usage in LocalStorageAllocateFlow

add DiskAO supports in LocalStorageAllocateCapacityFlow.

If the DiskAO.primaryStoreUuid of the root disk is specified,
Directly use DiskAO.primaryStorageUuid as the primary storage
for the boot disk in LocalStorageAllocateCapacityFlow.

Resolves: ZSV-8585

Change-Id: I696a68786d69686c6a626970627076666a776376
